### ISSUE TITLE
plantuml support offline mode

### ISSUE THREAD

**Issue by xulongzhe** [marktext/marktext#2995]
### Steps to reproduce

1. [First step]
2. [Second step]
3. [and so on...]

**Expected behavior:**

**Actual behavior:**

**Link to an example: [optional]**

### Versions

- MarkText:
- Operating system:

**Comment by fxha** [comment]
I agree that a settings option to change the PlantUML server url should be implemented.

**Comment by xwyam** [comment]
+1, for adding an option to change plantuml server url

**Comment by zebin-code** [comment]
Is this feature implemented?

**Comment by Jocs** [comment]
Thanks for the request! This is now supported. MarkText has a configurable PlantUML server URL in Preferences -> Markdown -> Diagrams (the `plantumlServer` setting, default `https://www.plantuml.com/plantuml`). You can point it at a self-hosted PlantUML server running on your offline network (for example a local plantuml-server built on plantuml.jar), and diagrams will render against that instead of the public server. The renderer wires this through to the diagram engine, so no internet access is required once your internal server is configured. Closing as implemented — please reopen with details if your offline setup still doesn't work for you.
